### ISSUE TITLE
[BUG] TypeError: has_permission() got an unexpected keyword argument 'scope'

### ISSUE THREAD

**Issue by darahsten** [NilCoalescing/djangochannelsrestframework#148]
**Describe the bug**
This is ideally a hybrid (feature request/bug/compatibility) depending on the authors' intent.

I was following the tutorial and at the point of implementing the chat in `chat/room.html`. I was getting a socket closed because
of the underlying problem.

This is a compatibility issue, by default, this system is designed to use the rest_frame work settings (including permission settings).

For example [IsAuthenticated Permission](https://github.com/encode/django-rest-framework/blob/master/rest_framework/permissions.py#L131). Its has permissions method expects 3 parameters, self, request and view. In contrast, the [has_permissions](https://github.com/NilCoalescing/djangochannelsrestframework/blob/master/djangochannelsrestframework/permissions.py#L113) of the channelsrestframework expects the fields scope, among others, and which by default will be provided by the calling functions. 
By default the has permissions method will come from the default permission classes defined in the rest_framework permissions will do not expect and do not receive scope as a field hence the above error.
Possible solutions include.
Specify alternate default permission classes for the consumers used by djangochannelsrest framework
 e.g. **CHANNELS_DEFAULT_PERMISSION_CLASSES** and give those the sensible defaults in this case
````
class AsyncAPIConsumer(...):
     permission_classes = djangochannelsrestframework.permissions.IsAuthenticated // as an example 
     **OR**
     permission_classes = api_settings.CHANNELS_DEFAULT_PERMISSION_CLASSES
```
This would be augmented by the documentation specifying the same issue.

**To Reproduce**
1. Follow the tutorial as in https://djangochannelsrestframework.readthedocs.io/en/latest/tutorial/part_1.html + part_2.html
This can be equivalent to just defining the Consumer.
```
class RoomConsumer(ObserverModelInstanceMixin, GenericAsyncAPIConsumer):
    queryset = models.Room.objects.all()
    serializer_class = serializers.ChatRoomSerializer
    lookup_field = "pk"
  # note permission_classes not defined. It may be possible to solve this by setting custom permission classes
  # per consumer
   ...
```

3. Use any default rest_frame work permission settings.

```
REST_FRAMEWORK = {
   ...,
    "DEFAULT_PERMISSION_CLASSES": ('rest_framework.permissions.AllowAny', ),
     ...
}.
```.


**LOG**
```
INFO 2022-07-05 19:49:31,253 runserver 171171 140189251778304 HTTP GET /static/images/favicons/favicon.ico 304 [0.00, 127.0.0.1:48004]
ERROR 2022-07-05 19:49:31,517 server 171171 140189251778304 Exception inside application: has_permission() got an unexpected keyword argument 'scope'
Traceback (most recent call last):
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/security/websocket.py", line 37, in __call__
    return await self.application(scope, receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/routing.py", line 150, in __call__
    return await application(
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/consumer.py", line 73, in dispatch
    await handler(message)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/generic/websocket.py", line 194, in websocket_receive
    await self.receive(text_data=message["text"])
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/generic/websocket.py", line 257, in receive
    await self.receive_json(await self.decode_json(text_data), **kwargs)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/djangochannelsrestframework/consumers.py", line 181, in receive_json
    await self.handle_action(action, request_id=request_id, **content)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/djangochannelsrestframework/consumers.py", line 176, in handle_action
    await self.handle_exception(exc, action=action, request_id=request_id)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/djangochannelsrestframework/consumers.py", line 136, in handle_exception
    raise exc
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/djangochannelsrestframework/consumers.py", line 157, in handle_action
    await self.check_permissions(action, **kwargs)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/djangochannelsrestframework/consumers.py", line 112, in check_permissions
    if not await ensure_async(permission.has_permission)(
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/asgiref/sync.py", line 435, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "/usr/lib/python3.9/asyncio/tasks.py", line 442, in wait_for
    return await fut
  File "/usr/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/channels/db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "/home/user/.virtualenvs/kaddle/lib/python3.9/site-packages/asgiref/sync.py", line 476, in thread_handler
    return func(*args, **kwargs)
TypeError: has_permission() got an unexpected keyword argument 'scope'
```

 - channelrestframework  1.0.0
 - channels 3.0.5 
 - django > 3.0

**Comment by hishnash** [comment]
You quite right, I think what we should do is when falling back to the default (rest framework permissions) we should call them in such a way as to ensure we do not pass scope but pass a proxy object request object in the same way rest framework methods expect. I suppose the best way to do this is to check if the permissions class subclasses a given subclass or not.

**Comment by darahsten** [comment]
Okay, that is even a much better solution.

**Comment by hishnash** [comment]
In the end I due to some other bugs I opted to change things so we no longer used any settings from the `REST_FRAMEWORK` domain.

**Comment by hishnash** [comment]
Fixed in v1.1 will no longer retrieve values from REST_FRAMEWORK settings.
